homeworks/src/homework12/Box.java

```diff
@@ -0,0 +1,46 @@
+package homework12;
+
+public class Box<T extends Fruit> {
+    private int amount;
+    private T kind;
```

Comment:
in general, the task is to use generics, so the right way will be using List<T> fruits instead of kind + amount and operate with it in all methods